File: src/Web/Spa/src/app/models/assignment.ts
import { Class } from './class';
import { AssignmentType } from './assignmentType';

export class Assignment{
    id: number;
    class: Class;
    name: string;
    type: AssignmentType;
    description: string;
    deadlineDateTime: Date;
    weight: number;

    // public DueTime?:Date;
    // public Name?:string;
    // public CourseName?:string;
    // public Description?:string;

    // public constructor(name:string, dueTime:Date,courseName:string,description:string){
    //     this.DueTime=dueTime;
    //     this.Name=name;
    //     this.CourseName=courseName;
    //     this.Description=description;
    // }
}

File: src/Services/Course/Course.Common/Models/Assignment.cs
using System;
using System.Collections.Generic;

namespace Course.Common.Models
{
    public class Assignment
    {
        public int Id { get; set; }

        public string ClassId { get; set; }
        public int ClassSemester { get; set; }
        public int ClassYear { get; set; }
        public string ClassCourseId { get; set; }
        public Class Class { get; set; }

        public AssignmentType Type { get; set; }
        public string Description { get; set; }
        public DateTime DeadlineDateTime { get; set; }
        public int Weight { get; set; }

        public List<StudentEvaluation> StudentEvaluations { get; set; }
        public List<GroupEvaluation> GroupEvaluations { get; set; }
        public List<AssignmentAttachment> AssignmentAttachments { get; set; }
    }
}
